### PR TITLE
fix(engine/test): Test inheritance contact group 

### PR DESCRIPTION
**Pull request by sechkem**
-Add a robot test to check the full compatibility of the contactgroup inheritance between the new engine configuration and the old one.

REFS: MON-151622

### CODE REVIEW

File: tests/engine/contactgroup_inheritance.robot [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+    Ctn Engine Config Delete Key In Cfg    0    contactgroup_1    alias    contactgroups.cfg
+    Ctn Engine Config Delete Key In Cfg    0    contactgroup_1    members    contactgroups.cfg
+    
+    # Set contactgroup_1 to use contactgroup_template_1
+    Ctn Engine Config Set Key Value In Cfg    0    contactgroup_1    use    contactgroup_template_1    contactgroups.cfg
+
+    # Operation in contactTemplates
+    ${config_values}    Create Dictionary
+...    contactgroup_members    contactgroup_3
+...    members    U1
```
Comment (bouda1) [inline tests/engine/contactgroup_inheritance.robot:56]: again the identation here is bad.

File: tests/engine/contactgroup_inheritance.robot [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+    Ctn Add Contact Group    ${0}    ${2}    ["U1"]
+    Ctn Add Contact Group    ${0}    ${3}    ["U2"]
+    Ctn Add Contact Group    ${0}    ${4}    ["U3","U4"]
+
+    # Contactgroup_1 to use contactgroup_template_1
+    Ctn Engine Config Set Key Value In Cfg    0    contactgroup_1    use    contactgroup_template_1    contactgroups.cfg
+
+    # Operation in contactTemplates
+    ${config_values}    Create Dictionary
+...    contactgroup_members    contactgroup_2
```
Comment (bouda1) [inline tests/engine/contactgroup_inheritance.robot:117]: same here

File: tests/engine/contactgroup_inheritance.robot [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+
+    FOR    ${key}    ${value}    IN    &{config_values}
+        Ctn Engine Config Set Key Value In Cfg    0    contactgroup_1    ${key}    ${value}    contactgroups.cfg
+    END
+
+    ${config_values_tmp}    Create Dictionary
+...    contactgroup_members    contactgroup_4
+...    members    U2
```
Comment (bouda1) [inline tests/engine/contactgroup_inheritance.robot:125]: same here

File: tests/engine/contactgroup_inheritance.robot [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+    Ctn Engine Config Delete Key In Cfg    0    contactgroup_1    alias    contactgroups.cfg
+    Ctn Engine Config Delete Key In Cfg    0    contactgroup_1    members    contactgroups.cfg
+    
+    # Contactgroup_1 to use contactgroup_template_1
+    Ctn Engine Config Set Key Value In Cfg    0    contactgroup_1    use    contactgroup_template_1    contactgroups.cfg
+
+    # Operation in contactTemplates
+    ${config_values}    Create Dictionary
+...    contactgroup_members    contactgroup_3
+...    members    U1
```
Comment (bouda1) [inline tests/engine/contactgroup_inheritance.robot:199]: and heere

File: tests/engine/contactgroup_inheritance.robot [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+    # Delete unnecessary fields in contactgroup:
+    Ctn Engine Config Delete Key In Cfg    0    contactgroup_1    members    contactgroups.cfg
+
+    # Contactgroup_1 to use contactgroup_template_1
+    Ctn Engine Config Set Key Value In Cfg    0    contactgroup_1    use    contactgroup_template_1    contactgroups.cfg
+
+    # Operation in contactgroup
+    ${config_values}    Create Dictionary
+...    contactgroup_members    contactgroup_3
+...    members    U1
```
Comment (bouda1) [inline tests/engine/contactgroup_inheritance.robot:279]: here

File: tests/engine/contactgroup_inheritance.robot [hunk 6]
```diff
@@ -0,0 +1,318 @@
...
+
+    FOR    ${key}    ${value}    IN    &{config_values}
+        Ctn Engine Config Set Key Value In Cfg    0    contactgroup_1    ${key}    ${value}    contactgroups.cfg
+    END
+
+    # Operation in contactgroupTemplates
+    ${config_values_tmp}    Create Dictionary
+...    contactgroup_members    contactgroup_4
+...    members    John_Doe
```
Comment (bouda1) [inline tests/engine/contactgroup_inheritance.robot:288]: again

File: tests/engine/contactgroup_inheritance.robot [hunk 7]
```diff
@@ -0,0 +1,318 @@
...
+    # Delete unnecessary fields in templates:
+    Ctn Engine Config Delete Key In Cfg    0    contactgroup_template_1    active_checks_enabled    contactgroupTemplates.cfg
+    Ctn Engine Config Delete Key In Cfg    0    contactgroup_template_1    passive_checks_enabled    contactgroupTemplates.cfg
+
+    # Add Command :
+    Ctn Engine Config Add Command    0    command_notif    /usr/bin/true
+    Ctn Engine Config Set Value In Contacts    0    John_Doe    host_notification_commands    command_notif
+    Ctn Engine Config Set Value In Contacts    0    John_Doe    service_notification_commands    command_notif
+
+    # Add necessarily files :
```
Comment (bouda1) [inline tests/engine/contactgroup_inheritance.robot:101]:
```suggestion
    # Add the necessary files :
```

File: tests/engine/contactgroup_inheritance.robot (hunk 1 above)
Comment (sechkem) [reply tests/engine/contactgroup_inheritance.robot:56]: will fix

**Review by bouda1 — APPROVED** [review]

**Pull request merged**